File: lib/to_mdx/parsers/open_document/text/line_break.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Text
      class LineBreak < Base
        def initialize(element = nil, context = {})
          super(element, context)
        end

        def empty?
          false
        end

        def to_mdx
          "<br />"
        end

        def to_s
          "\n"
        end

        register_for "text:line-break"
      end
    end
  end
end

File: src/parsers/open-document/text/line_break.ts
import { BaseElement } from "../../base_element.ts";

export class LineBreak extends BaseElement {
  empty(): boolean {
    return false;
  }

  toMdx(): string {
    return "<br />";
  }

  toString(): string {
    return "\n";
  }

  static {
    BaseElement.registerFor.call(this, "text:line-break");
  }
}
